<a href="https://colab.research.google.com/github/kadiatou12/data_analytics_2/blob/main/Copy_of_Copy_of_Unit3_Python_Advanced_Pandas_DataWrangling_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment 1 : Python Data Analysis using salaries dataset

# Data Exploration:  This should include summary statistics, means, medians, quartiles, or any other relevant information about the data set. 

In [ ]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [ ]:
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')

In [ ]:
type(df)

pandas.core.frame.DataFrame

In [ ]:
# 397 rows = 397 employees
df.shape

(397, 7)

In [ ]:
df.head()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
0           1      Prof          B             19           18  Male  139750
1           2      Prof          B             20           16  Male  173200
2           3  AsstProf          B              4            3  Male   79750
3           4      Prof          B             45           39  Male  115000
4           5      Prof          B             40           41  Male  141500

In [ ]:
df.tail()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
392         393      Prof          A             33           30  Male  103106
393         394      Prof          A             31           19  Male  150564
394         395      Prof          A             42           25  Male  101738
395         396      Prof          A             25           15  Male   95329
396         397  AsstProf          A              8            4  Male   81035

In [ ]:
df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  397.000000     397.000000   397.000000     397.000000
mean   199.000000      22.314861    17.614610  113706.458438
std    114.748275      12.887003    13.006024   30289.038695
min      1.000000       1.000000     0.000000   57800.000000
25%    100.000000      12.000000     7.000000   91000.000000
50%    199.000000      21.000000    16.000000  107300.000000
75%    298.000000      32.000000    27.000000  134185.000000
max    397.000000      56.000000    60.000000  231545.000000

# Mean/median of all salaries

In [ ]:
# Mean of all salaries


In [ ]:
# Median of all salaries


# Median salary between gender and/or rank(title) Etc...


Making a new Data Frame that only has the columns 'sex' and 'salary'

In [ ]:
#df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')
new_df = df[['sex','salary']]


In [ ]:
new_df.head()

sex  salary
0  Male  139750
1  Male  173200
2  Male   79750
3  Male  115000
4  Male  141500

Finding the Median salary for all males

In [ ]:
male_df = new_df[(new_df['sex'] == 'Male')]

In [ ]:
male_df['salary'].median()

108043.0

Finding the Median salary for all females

In [ ]:
female_df = new_df[(new_df['sex'] == 'Female')]

In [ ]:
female_df['salary'].median()

103750.0

Comparing the Mean salary of Males and Females

In [ ]:
male_mean = (new_df[new_df['sex'] == 'Male']).mean()
print(male_mean.to_string())
#male_df = new_df[(new_df['sex'] == 'Male')]
#male_df['salary'].mean()


salary    115090.418994


In [ ]:
female_mean = (new_df[new_df['sex'] == 'Female']).mean()
print(female_mean.to_string())

salary    101002.410256


In [ ]:
# The difference between mean salaries of male and female
compare = male_mean- female_mean
print(compare.to_string())

salary    14088.008738


# Data Wrangling – perform some basic transformations on the data

People who have 20 years of experience (Total: 157), (Mean Salary: 122,520.66)

In [ ]:
ex_sal = df[['yrs.service','salary']]
print(ex_sal)

     yrs.service  salary
0             18  139750
1             16  173200
2              3   79750
3             39  115000
4             41  141500
..           ...     ...
392           30  103106
393           19  150564
394           25  101738
395           15   95329
396            4   81035

[397 rows x 2 columns]


In [ ]:
Ex_20 = ex_sal[ex_sal['yrs.service']>=20]
print(Ex_20.count())

yrs.service    157
salary         157
dtype: int64


In [ ]:
print((Ex_20).mean())

yrs.service        31.057325
salary         122520.656051
dtype: float64


Comparing the salaries of people who have 20+ years of experience vs less than 20 years of experience

In [ ]:
Ex_less_20 = ex_sal[ex_sal['yrs.service']<20]

In [ ]:
print((Ex_less_20).mean())

yrs.service         8.820833
salary         107940.504167
dtype: float64


In [ ]:
diff = Ex_20.mean()[['salary']] - Ex_less_20.mean()[['salary']]
print(diff.to_string())

salary    14580.151884


People who have salary of greater than or equal to 200,000

In [ ]:
grater_200k = ex_sal[ex_sal['salary']>=200000]
print(grater_200k.count().to_string())

yrs.service    3
salary         3


## **Conclusion**

*   We can say that in this that Median salary for all females is 103750.0, and all males is 108043.0
*   The difference in mean salaries of male and female is 14088.01
*   Employees gets $14580.15 more salary if they have  experinence over 20 years
*   3 Employees have salary of greater than or equal to 200,000




# Rename values, subset columns, etc

# Create some simple visualizations showing some simple statistics

Your observations from the visuals: 

# Draw some conclusions based on the data.  What did you discover?

Based on the analysis performed above, the conclusions are as follows:

Your Findings here...



I also discovered that the -

 Your Findings here...



